fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/BaseProtocolMetricMaker.java

```diff
@@ -59,12 +61,68 @@ public LogicalMetric makeInnerWithResolvedDependencies(
             LogicalMetricInfo logicalMetricInfo,
             List<LogicalMetric> dependentMetrics
     ) {
-        TemplateDruidQuery partialQuery = makePartialQuery(logicalMetricInfo, dependentMetrics);
-        ResultSetMapper calculation = makeCalculation(logicalMetricInfo, dependentMetrics);
-        ProtocolSupport protocolSupport = makeProtocolSupport(logicalMetricInfo, dependentMetrics);
+        LogicalMetric dependentMetric = dependentMetrics.get(0);
+        LogicalMetric renamedDependentMetric = renameIfConflicting(logicalMetricInfo.getName(), dependentMetric);
+        List<LogicalMetric> renamedDependentMetrics =
+                dependentMetrics.size() == 1 ? Collections.singletonList(renamedDependentMetric) : dependentMetrics;
+        TemplateDruidQuery partialQuery = makePartialQuery(logicalMetricInfo, renamedDependentMetrics);
+        ResultSetMapper calculation = makeCalculation(logicalMetricInfo, renamedDependentMetrics);
+        ProtocolSupport protocolSupport = makeProtocolSupport(logicalMetricInfo, renamedDependentMetrics);
         return new ProtocolMetricImpl(logicalMetricInfo, partialQuery, calculation, protocolSupport);
     }
 
+    /**
+     * Renames the provided logical metric if there is a name conflict between it and the final desired output name.
+     * If there is no name conflict the input metric is returned unchanged. Otherwise, the input metric is renamed to
+     * not conflict with the final output name.
+     * <p>
+     * This method specifically the {@link GeneratedMetricInfo} typing on the input metric if it has to be renamed. All
```

Comment:
I think there's a grammar error here, but I'm not exactly sure what you were trying to say in this sentence.